### ISSUE TITLE
Chat messaging list not showing avatar

### ISSUE THREAD

**Issue by savked** [GetStream/stream-chat-android#291]
Related to issue #253 

I did change the list of channels to our old looking style, however it is not yet confirmed if we are going to keep GetStream chat list activity or try and reimplement our old one from Layer library.

In the gif provided you can see that at first there is no avatar showing for either of users, but when I send the message from my end, you can see my avatar showing for a second and then disappearing again. What the issue could be here.

https://imgur.com/a/ziHumQc

NOTE: Using the latest version 3.2.3

**Comment by elevenetc** [comment]
We're using `Glide` internally to load pictures and nothing else really happen. My assumption from your gif is that picture was loaded and showed, but then on another layout pass Glide tried to load it again and failed to do so: maybe because of network issue or because of memory constraints. 

I will create SDK release with extra logs for you, so we can see what actually happens within Glide.

@savked And by the way, do you have that image file? Whats size of the file?

**Comment by savked** [comment]
@elevenetc here is the image that the chat is loading:

https://tourlina-staging.imgix.net/2017/03/02/10/23/51/f36e8f3f-0b84-49db-844c-9a3470350e58/16864972_163598174147012_3601214643003946294_n.jpg

**Comment by elevenetc** [comment]
@savked could you try this release `3.2.3-glide-logs`? And see logcat logs with tag `Glide` when image is not loaded.

**Comment by savked** [comment]
This is the only thing I get: 2019-11-12 16:58:40.410 26442-26442/de.tourlina.Tourlina.staging W/Glide: Failed to find GeneratedAppGlideModule. You should include an annotationProcessor compile dependency on com.github.bumptech.glide:compiler in your application and a @GlideModule annotated AppGlideModule implementation or LibraryGlideModules will be silently ignored

**Comment by savked** [comment]
Adding the glide annotation processor (annotationProcessor 'com.github.bumptech.glide:compiler:4.9.0') does not fix the problem.

**Comment by elevenetc** [comment]
Hm, are you sure that you're using `3.2.3-glide-logs` version? That's the logs from old version of SDK where Glide logs are disable, so `Failed to find GeneratedAppGlideModule` is valid message.

The `3.2.3-glide-logs` version has `AppGlideModule`: https://github.com/GetStream/stream-chat-android/commit/038280d74bdf15172bf700592fd878c9b5a8ddb8
```groovy
implementation 'com.github.GetStream:stream-chat-android:3.2.3-glide-logs'
```
I just created a sample app from scratch and see the logs:
```
Glide  D  Discovered AppGlideModule from annotation: com.getstream.sdk.chat.utils.images.StreamGlideModule
       D  Finished loading Bitmap from REMOTE for https://www.fujifilm.com/products/digital_cameras/x/fujifilm_x_t3/sample_images/img/index/ff_x_t3_002.JPG with size [2146x2146] in 2686.777655 ms
```

**Comment by savked** [comment]
I am using the given verison. Next to the "Failed to find GeneratedAppGlideModule" I managed to get one other output like:

2019-11-12 20:49:57.282 7951-7951/de.tourlina.Tourlina.staging D/Glide: Finished loading BitmapDrawable from REMOTE for https://tourlina-staging.imgix.net/2019/11/08/09/00/52/46b68322-5829-4fcc-9d97-3f5b4872c37b/16508845_149732622201883_1126592597476874017_n.jpg with size [112x112] in 536.9647 ms

And this image is from my device (currently logged user).

However, that's when chat managed to load this image. After that, I tried re-entering the ChatFragment, but only once got it working.

**Comment by elevenetc** [comment]
So it means you're getting logs. You can also try to see whats logged with `Engine` tag ( you need to increase log level for it: `adb shell setprop log.tag.Engine VERBOSE` ).

I'm afraid the issue is somewhere outside of SDK: we're not replacing avatar view and depend completely on Glide. So it either loads image or not. If not there should be error message. And as I said before another possible reason is memory constraints and here is [Glide docs quote](https://bumptech.github.io/glide/doc/debugging.html#out-of-memory-errors):
```
Almost all OOM errors are due to issues with the hosting application and not with Glide.
```
I also build another SDK version with debuggable avatar views: `3.2.3-glide-logs-alpha-1`. It has just `ImageView` and show `Glide` code:
```java
private void configAvatar(String image, String initial) {

    ImageView imageView = new ImageView(context);
    imageView.setBackgroundColor(Color.RED);

    if (!Utils.isSVGImage(image))
        Glide.with(context)
                .load(StreamChat.getInstance(context).getUploadStorage().signGlideUrl(image))
                .apply(RequestOptions.circleCropTransform())
                .into(imageView);

    RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(
            (style.getAvatarWidth()),
            (style.getAvatarHeight()));
    imageView.setLayoutParams(params);

    addView(imageView);
}
```
If you still have problem (avatars are red squares) then I would try debug and review memory consumption as `Glide` docs recommend.

**Comment by savked** [comment]
Thank you for your hard work @elevenetc.

I just tried the '3.2.3-glide-logs-alpha-1' version and the avatars are red squares.

I noticed that even when the image loads, that's for the current user (message to the right) but for the other user it is never showing.

When I see the response from channel list I get this one channel that has this two users in it, however I noticed that under members there is no `image` field, is there supposed to be one, could this be a back-end error on our side?

Here is the response "members":

`         "members":[
            {
               "user":{
                  "id":"181",
                  "role":"user",
                  "created_at":"2019-09-17T10:13:45.485842Z",
                  "updated_at":"2019-11-13T15:10:21.865985Z",
                  "last_active":"2019-11-13T15:08:04.475301Z",
                  "online":false,
                  "name":"Petar"
               },
               "role":"member",
               "created_at":"2019-11-08T12:14:08.139872Z",
               "updated_at":"2019-11-08T12:14:08.139872Z"
            },
            {
               "user":{
                  "id":"182",
                  "role":"user",
                  "created_at":"2019-09-17T10:14:45.713314Z",
                  "updated_at":"2019-11-13T15:10:44.643188Z",
                  "last_active":"2019-11-13T15:05:46.173816Z",
                  "online":true,
                  "name":"Ivana"
               },
               "role":"member",
               "created_at":"2019-11-08T12:14:08.139872Z",
               "updated_at":"2019-11-08T12:14:08.139872Z"
            }`

**Comment by tbarbugli** [comment]
@savked how did you log the response?

**Comment by savked** [comment]
@tbarbugli Hello Tommaso, I got the response when entering my ChatFragment that's when the channels query is called.

2019-11-13 17:01:59.633 10290-10967/de.tourlina.Tourlina.staging D/OkHttp: <-- 200 OK https://chat-us-east-1.stream-io-api.com/channels?api_key=**[OUR API KEY HERE]**&user_id=182&client_id=19883ff2-7e92-44fd-9158-391f7dc78add&payload=%7B%22filter_conditions%22%3A%7B%22%24and%22%3A%5B%7B%22type%22%3A%22messaging%22%7D%5D%7D%2C%22sort%22%3A%5B%5D%2C%22presence%22%3Afalse%2C%22state%22%3Atrue%2C%22watch%22%3Atrue%7D (668ms)

And the members field I posted above is just the part of the response.

**Comment by tbarbugli** [comment]
I think the problem is not on the response payload, APIs include extra data for members.user. The issue is that we do not deserialize JSON correctly. I created a branch with failing tests (gson-user)

**Comment by savked** [comment]
Alright, thank you, please update on how this is going and if necessary ask for more info.

Also, canln I ask you to check your email @tbarbugli ?

**Comment by elevenetc** [comment]
@savked would be useful if you record full `OkHttp` logs from app start, something like:
```
pidcat com.your.package -t OkHttp > stream-okhttp-logs.txt
```
And by the way, do you have the same issue on iOS? If you have it of course.

**Comment by savked** [comment]
@elevenetc I am not able to "record" this log, nothing happens, an emtpy file is made with the device name in it and thats it. I did not have much time to tinker with this.

Our iOS version is using the old layout so it's pulling data from other sources.

However, I noticed that all of the sudden the other users image is showing, but the current one is still not. Is this something was done on your end or maybe on ours (back-end)? I did not update the library.

**Comment by elevenetc** [comment]
> Is this something was done on your end or maybe on ours (back-end)?

Nothing was fixed on backend.
> but after couple of times closing and opening the app again, it stopped working.

This behaviour again reminds me memory a constraints issue. I'm not able to reproduce the issue on Sample app with even low end device with small amount of memory. And we don't have such kind of issues on iOS or Android with other apps.

So to help with your issue I need to have either more logs ( Glide, garbage collector or eventually the whole log of you app ) or more context that I'm able to reproduce the issue.

**Comment by savked** [comment]
@tbarbugli mention that it's some sort of a deserialization problem on your end, have you looked into this?

Can I send you the log somewhere more privately?

**Comment by elevenetc** [comment]
Yeah, we checked deserialisation, is works as expected. And that's why I asked about more logs, would be nice to see the whole response where we can see absence of image fields.

You can send me logs to [eugene@getstream.io](mailto:eugene@getstream.io?subject=github-issue).

**Comment by savked** [comment]
Sent to @elevenetc 

However I definitely noticed that leaving one device inside the app, sleep or awake, on other device in chat the avatar will show, there is definitely no problem with memory, but something back-end related.

**Comment by savked** [comment]
Since I am not using the avatars inside the chat which can be seen in issue #321 is there any way that I can explicitly set the avatar image to the url or file I provide?

There is nothing related to this in documentation here: https://github.com/GetStream/stream-chat-android/blob/master/docs/ChannelHeader.md

**Comment by savked** [comment]
Nevermind, I have changed the layout and header view, you can close this issue.

**Comment by elevenetc** [comment]
Hello @savked 

We’ve made some investigations and in our backend database your users have `image` fields, but they are null. We’re not removing or deleting users data. So my next guess is that it was set to null from client side.

How do you set user images?
Could it be something like this?
```java
extraData.put("name", "Petar");
extraData.put("image", null);
User user = new User("user-id", extraData);
client.setUser(user);
```

**Comment by savked** [comment]
This is how I do it, however picture before I change it is like this:

`https://tourlina-staging.imgix.net/2017/03/02/10/23/51/f36e8f3f-0b84-49db-844c-9a3470350e58/16864972_163598174147012_3601214643003946294_n.jpg?ixlib=rb-3.1.0&w=1200&h=1200&fit=crop&crop=faces`

Where I need to change this last part or otherwise Glide won't load the image:

`https://tourlina-staging.imgix.net/2017/03/02/10/23/51/f36e8f3f-0b84-49db-844c-9a3470350e58/16864972_163598174147012_3601214643003946294_n.jpg`

     private void configureStreamClient() {
        int userId = CurrentUserManager.getInstance().getUser(getActivity().getCacheDir()).getId();

        Client client = StreamChat.getInstance(getActivity().getApplication());
        HashMap<String, Object> extraData = new HashMap<>();
        extraData.put("name", CurrentUserManager.getInstance().getUser(getActivity().getCacheDir()).getName());
        String userPicture = CurrentUserManager.getInstance().getUser(getActivity().getCacheDir()).getPicture();
        // Images gathered from the server have some ending like "?ixlib=rb-3.1.0&w=150&h=150&fit=crop&crop=faces" that library would not accept as an image, therefore we are doing an adjustments
        userPicture = userPicture.substring(0, userPicture.indexOf("?"));
        extraData.put("image", userPicture);
        User currentUser = new User(String.valueOf(userId), extraData);
        String token = CurrentUserManager.getInstance().getUser(getActivity().getCacheDir()).getStream_chat_token();

        client.setUser(currentUser, token);
    }

Which means there is no way image is null.

**Comment by elevenetc** [comment]
Hello @savked , checked backend again and we can't find any place where image fields could removed, except client initialisation. Android SDK allowed to set null field, so this is the only place where user image could be replaced with null.

We fixed it and published new SDK release: `3.4.1`. It ignores null image field.

Please update SDK and let us know if there is still the same issue.
